components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m

```diff
@@ -481,6 +508,91 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
   [self.view setNeedsLayout];
 }
 
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^_Nullable)(BOOL))completion {
+  CGSize newPreferredContentSize =
+      CGSizeMake(CGRectGetWidth(self.view.bounds), preferredContentHeight);
+  if ([self isMobileLandscape]) {
+    self.contentViewController.preferredContentSize = newPreferredContentSize;
+    return;
+  }
+  CGPoint contentOffset =
+      [self calculateContentOffsetWithPreferredContentHeight:preferredContentHeight];
+  CGFloat precentageOfFullScreen =
+      [self precentageOfFullScreenWithPreferredContentHeight:preferredContentHeight];
+  BOOL shouldSetContentHeight = NO;
+  if (_contentVCPreferredContentSizeHeightCached > preferredContentHeight) {
```

Comment:
`_contentVCPreferredContentSizeHeightCached` is the old preferredContentHeight, if that is larger than what we need we will always have content on the bottom so we can animate down and set this later. If this is smaller than the new height then we need to set this now so we don't see the scrim as we animate.